contactsummary/tests/phpunit/CRM/Contactsummary/Utils/ContractSummaryTestTrait.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * A trait with helper methods to be reused among this extension's tests
+ */
+trait ContractSummaryTestTrait {
+
+  /**
+   * Creates absence period from specified start and end dates and return its ID
+   *
+   * @param $startDate
+   * @param $endDate
+   *
+   * @return int
+   */
+  protected function createAbsencePeriod($startDate, $endDate)  {
+    $params = array('name'=> $startDate,'start_date'=>$startDate, 'end_date'=>$endDate);
+    $period = civicrm_api3('HRAbsencePeriod', 'create', $params);
+    return $period['id'];
+  }
+
+  /**
+   * Creates absence entitlements for specific leave type
+   *
+   * @param $params
+   *
+   * @return int
+   */
+  protected function createAbsenceEntitlement($params)  {
+    $period = CRM_HRAbsence_BAO_HRAbsenceEntitlement::create($params);
```

Comment:
Why are you using the API to create the Absence Period and here you're using the BAO?